src/frontend/src/lib/components/signer/SignerConsentMessage.svelte

```diff
@@ -0,0 +1,153 @@
+<script lang="ts">
+	import { Markdown } from '@dfinity/gix-components';
+	import type {
+		icrc21_consent_info,
+		ConsentMessageApproval,
+		Rejection
+	} from '@dfinity/oisy-wallet-signer';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { getContext } from 'svelte';
+	import { fade } from 'svelte/transition';
+	import SignerLoading from '$lib/components/signer/SignerLoading.svelte';
+	import SignerOrigin from '$lib/components/signer/SignerOrigin.svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { SIGNER_CONTEXT_KEY, type SignerContext } from '$lib/stores/signer.store';
+	import { toastsError } from '$lib/stores/toasts.store';
+
+	const {
+		consentMessagePrompt: { payload, reset: resetPrompt },
+		callCanisterPrompt: { reset: resetCallCanisterPrompt }
+	} = getContext<SignerContext>(SIGNER_CONTEXT_KEY);
+
+	let approve: ConsentMessageApproval | undefined;
+	$: approve = $payload?.status === 'result' ? $payload?.approve : undefined;
+
+	let reject: Rejection | undefined;
+	$: reject = $payload?.status === 'result' ? $payload?.reject : undefined;
+
+	let consentInfo: icrc21_consent_info | undefined;
+	$: consentInfo = $payload?.status === 'result' ? $payload?.consentInfo : undefined;
+
+	let loading = false;
+	let displayMessage: string | undefined;
+
+	const onPayload = () => {
+		if ($payload?.status === 'loading') {
+			displayMessage = undefined;
+			loading = true;
+
+			// In case the relying party has not closed the window between the last call and requesting a new call, we reset the call canister prompt; otherwise, we might display both the previous result screen and the consent message.
+			// Note that the library handles the case where the relying party tries to submit another call while a call is still being processed. Therefore, we cannot close a prompt here that is not yet finished.
+			resetCallCanisterPrompt();
+
+			return;
+		}
+
+		loading = false;
+
+		if ($payload?.status === 'error') {
+			toastsError({
+				msg: { text: $i18n.signer.consent_message.error.retrieve },
+				err: $payload.details
+			});
+			return;
+		}
+
+		displayMessage =
+			nonNullish(consentInfo) && 'GenericDisplayMessage' in consentInfo.consent_message
+				? consentInfo.consent_message.GenericDisplayMessage
+				: undefined;
+	};
+
+	$: $payload, onPayload();
+
+	type Text = { title: string; content: string } | undefined;
```

Comment:
shouldn't this be among the `types` folder?

Reply:
It's used nowhere else and really specific for this function and component therefore I would just keep it here.